lib/rubygems/util/list.rb

```diff
@@ -2,6 +2,8 @@ module Gem
   List = Struct.new(:value, :tail)
 
   class List
+    include Enumerable
```

Comment:
@byroot This isn't quote true. You have to remove Struct (which has its own Enumerable inclusion and meaning), and reverse the `to_a`. See PR #1200 ;)